lib/core/src/Cardano/Wallet.hs

```diff
@@ -2131,11 +2133,16 @@ guardJoin knownPools WalletDelegation{active,next} pid = do
 
 guardQuit
     :: WalletDelegation
+    -> Quantity "lovelace" Word64
     -> Either ErrCannotQuit ()
-guardQuit WalletDelegation{active,next} = do
+guardQuit WalletDelegation{active,next} rewards = do
     let last_ = maybe active (view #status) $ lastMay next
+
     unless (isDelegatingTo anyone last_) $
         Left ErrNotDelegatingOrAboutTo
+
+    unless (rewards == Quantity 0) $
```

Comment:
> We could have implemented some more
convoluted logic where the wallet would automatically collect rewards as
part of a de-registration, but we are a bit lacking time here so we'll
simply make this a manual operation that users have to undergo.

👌